pkg/services/vmoperator/vmopmachine.go

```diff
@@ -397,11 +428,14 @@ func (v *VmopMachineService) reconcileVMOperatorVM(ctx context.Context, supervis
 }
 
 func (v *VmopMachineService) reconcileNetwork(supervisorMachineCtx *vmware.SupervisorMachineContext, vm *vmoprv1.VirtualMachine) bool {
-	if vm.Status.VmIp == "" {
+	if vm.Status.Network.PrimaryIP4 == "" && vm.Status.Network.PrimaryIP6 == "" {
 		return false
 	}
 
-	supervisorMachineCtx.VSphereMachine.Status.IPAddr = vm.Status.VmIp
+	supervisorMachineCtx.VSphereMachine.Status.IPAddr = vm.Status.Network.PrimaryIP4
+	if supervisorMachineCtx.VSphereMachine.Status.IPAddr == "" {
```

Comment:
If IPv6 is new in vmop v1alpha2, should IPAddr in vSphereMachine also shows 2 IP addresses?

Reply:
I chose to implement the same behaviour for now like we had before when using v1alpha1.

I would keep API changes out of scope for now. If this is a feature we want, we should file a different issue for that.

Reply:
Sure. That 's out of this PR scope, please ignore it.